skyportal/handlers/api/candidate/scan_report.py

```diff
@@ -27,31 +25,17 @@ def get_infos_saved_sources_by_obj(session, group_ids, detection_range, saved_ra
 
     Returns
     -------
-    list of saved_infos, skyportal.models.Obj
+    list of tuples (obj_id, source_ids)
     """
     try:
-        return (
-            session.query(
-                Obj,
-                func.json_agg(
-                    func.distinct(
-                        func.json_build_object(
-                            "saved_at",
-                            Source.saved_at,
-                            "saved_by",
-                            User.username,
-                            "group",
-                            Group.name,
-                        ).cast(JSONB)
-                    )
-                ).label("saved_infos"),
+        return session.execute(
+            sa.select(
+                Source.obj_id,
+                sa.func.array_agg(sa.func.distinct(Source.id)).label("source_ids"),
```

Comment:
Not sure you need that honestly, could just get the object_id and handle anything else later when you build the record/item